common-pages/src/TimeFilter.tsx

```diff
@@ -0,0 +1,715 @@
+﻿//******************************************************************************************************
+//  TimeFilter.tsx - Gbtc
+//
+//  Copyright © 2020, Grid Protection Alliance.  All Rights Reserved.
+//
+//  Licensed to the Grid Protection Alliance (GPA) under one or more contributor license agreements. See
+//  the NOTICE file distributed with this work for additional information regarding copyright ownership.
+//  The GPA licenses this file to you under the MIT License (MIT), the "License"; you may not use this
+//  file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  09/16/2021 - Christoph Lackner
+//       Generated original version of source code.
+//  06/17/2024 - Ali Karrar
+//******************************************************************************************************
+
+import * as React from 'react';
+import moment from 'moment';
+import momentTZ from 'moment-timezone';
+import { DatePicker, Select, Input } from '@gpa-gemstone/react-forms'
+import { findAppropriateUnit, getMoment, getStartEndTime, units, ITimeFilter } from './TimeWindowUtils';
+
+interface IProps {
+    filter: ITimeFilter;
+    setFilter: (filter: ITimeFilter) => void,
+    showQuickSelect: boolean;
+    dateTimeSetting: 'center' | 'startWindow' | 'endWindow' | 'startEnd';
+    timeZone: string;
+    isHorizontal: boolean;
+}
+
+interface IQuickSelect { label: string, createFilter: (timeZone: string) => ITimeFilter }
+
+export const momentDateFormat = "MM/DD/YYYY";
+export const momentTimeFormat = "HH:mm:ss.SSS"; // Also is the gemstone format
+
+
+const AvailableQuickSelects: IQuickSelect[] = [
```

Comment:
I'd create a `TimeFilter` folder and move the `QuickSelects` into their own file